cmd/create.go

```diff
@@ -85,14 +86,14 @@ A full list of component types that can be deployed is available using: 'odo cat
 		// "Default" values
 		componentImageName := args[0]
 		componentType := args[0]
-		componentName := args[0]
+		componentName := util.ExtractComponentType(componentType)
```

Comment:
Shouldn't this be updated / modified in a different PR?

Reply:
This PR makes it possible to have component types as the following:
1. "myproject/python:3.5" -- Return python
2. "python:3.5" -- Return python
3. nodejs -- Return nodejs
So, it was essential to correctly identify the component type among other things(version, namespace) so, I added it as part of this PR

Reply:
Instead of writing this block twice, create one function which takes input and returns:
* componentImageName
* componentName
* componentType
* componentVersion

if not present, that function should fill whatever default values we have

Reply:
But what is the gain the 2 blocks will only move to a different function instead of this one right?